edgexfoundry/device-usb-camera
# feat: extend discovery for realsense cameras

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Make sure a realsense camera is plugged into your device.
1. Use `v4l2-ctl --list-devices` to check that the device is recognized by your system and to check the paths of the device.
1. Build the usb device service using `make docker`.
1. Run the edgex core services along with the device service.
1. Check the discovered devices (http://0.0.0.0:59881/api/v3/device/service/name/device-usb-camera), make sure that the paths match. Note that only the 1sat, 3rd and 5th paths are saved in the device, as these are the only streaming paths.
1. Test for rediscovery by deleting the device or restarting the service, etc.
1. Make sure the paths are updated if ports are changed.

## New Dependency Instructions (If applicable)
